cmd/web/config/config_test.go

```diff
@@ -532,22 +535,24 @@ func TestString_GetSensored(t *testing.T) {
 	cfg := Config{}
 
 	tests := []struct {
-		name string
-		c    Config
-		want string
+		name    string
+		c       Config
+		want    string
+		wantRet bool
 	}{
 		{
-			name: "Testing returned string",
-			c:    cfg,
-			want: cfg.String(),
+			name:    "Testing if string returned",
+			c:       cfg,
+			want:    cfg.String(),
```

Comment:
If your `want` keeps using the result of `cfg.String()` you will always get a successful test, when testing the result of `.String()` regardless if you use `strings.Contains()` or not. In order to have a reliable test, your `want` should be something you expect and should be a literal.

As an experiment, change `type (c *Config) String() string` to:
```go
func (c *Config) String() string {
	return "hakuna matata"
}
```

Your test will still pass. That makes it a bad test.

### What does the code do?
Alright. So let's look at what the String() method on `Config` does. It returns a "safe" string representation of the type. With _safe_ I mean that sensitive values are masked. So let's look at how it could be used.

### How can it be used?
Since Config implements the [Stringer](https://golang.org/pkg/fmt/#Stringer) interface (see https://play.golang.org/p/fyZz_LRY0Gm for an example), we can just print the type and we'll see a human-readable representation.

Ok cool, so this should work:
```go
cfg := Config{}
fmt.Printf("Our config looks like this: %s", cfg) // this implicitly calls `cfg.String()`, you can do it, but you don't need to. Pretty cool
// This will result in a very uninteresting output, but at least it compiles :-)
```

### How can we come up with a better test?
So, with this in mind, I think the test can be improved by doing the following:

- Replace the field `want string` with `shouldContain string`. To make it more clear that we expect the value to be contained in something else
- Set the value of `shouldContain` in the test parameters, to the constant value of `valueMask`
- Change the assertion to: `strings.Contains(tt.c.String(), tt.shouldContain)`

Then for a second test case:
- Update the config object, or create a new one and set a property, e.g.: `cfg.ListenOn = "localhost:12345"`
- shouldContain should be set to the literal value `"localhost:12345"`

This way you test that the two value are contained in the string representation. The _breaking_ change I made as an example will now fail the test.